File: GridView/RadGridViewColumn-Indicate-Editor-Type/RadGridViewColumn-indicate-editor-type-VB/IndicatedDateTimeColumn/IndicatedDateTimeColumn.vb
Imports Telerik.WinControls.UI

Public Class IndicatedDateTimeColumn
    Inherits GridViewDateTimeColumn

    Private _enableIndicator As Boolean

    Public Sub New()
        _enableIndicator = True
    End Sub

    Public Property EnableIndicator() As Boolean
        Get
            Return Me._enableIndicator
        End Get
        Set(ByVal value As Boolean)
            Me._enableIndicator = value
            Me.OwnerTemplate.Refresh()
        End Set
    End Property

    Public Overrides Function GetCellType(ByVal row As GridViewRowInfo) As Type
        If TypeOf row Is GridViewDataRowInfo Then
            Return GetType(IndicatedDateTimeCellElement)
        End If
        Return MyBase.GetCellType(row)
    End Function
End Class

File: GridView/GridViewIndicatedColumns/GridViewCustomCellsC#/IndicatedDateTimeColumn/IndicatedDateTimeColumn.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace GridViewCustomCells
{
    
    public class IndicatedDateTimeColumn : GridViewDateTimeColumn
    {
        private bool enableIndicator;

        public IndicatedDateTimeColumn()
        {
            enableIndicator = true;
        }

        public bool EnableIndicator
        {
            get
            {
                return this.enableIndicator;
            }
            set
            {
                this.enableIndicator = value;
                this.OwnerTemplate.Refresh();
            }
        }

        public override Type GetCellType(GridViewRowInfo row)
        {
            if (row is GridViewDataRowInfo)
            {
                return typeof(IndicatedDateTimeCellElement);
            }
            return base.GetCellType(row);
        }
    }
}